web/impact/impact/v1/views/base_list_view.py

```diff
@@ -32,18 +39,45 @@ def metadata(self):
         return result
 
     def get(self, request):
-        limit = int(request.GET.get('limit', 10))
-        offset = int(request.GET.get('offset', 0))
+        limit = self._validate_kwarg(
+            request.GET.get("limit", str(self.DEFAULT_LIMIT)), key="limit")
+        offset = self._validate_kwarg(
+            request.GET.get("offset", str(0)), key="offset")
+        if self.errors:
+            return Response(status=401, data=self.errors)
         base_url = _base_url(request)
         count, results = self.results(limit, offset)
         result = {
             "count": count,
-            "next": _url(base_url, limit, offset + limit),
-            "previous": _url(base_url, limit, offset - limit),
+            "next": _next_url(base_url, limit, offset, count),
+            "previous": _previous_url(base_url, limit, offset, count),
             "results": results,
         }
         return Response(result)
 
+    def _validate_kwarg(self, val, key):
```

Comment:
This would be simpler as two functions, say `_get_limit(request)` and `_get_offset(request)`